src/lambda/commands/createNewSamApp.ts

```diff
@@ -55,43 +54,47 @@ export async function resumeCreateNewSamApp(activationLaunchPath: ActivationLaun
 }
 
 export interface CreateNewSamApplicationResults {
-    reason: 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
-    result: 'pass' | 'fail' | 'cancel'
     runtime: string
+    result: result
 }
 
+type createReason = 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
+
 /**
  * Runs `sam init` in the given context and returns useful metadata about its invocation
  */
 export async function createNewSamApplication(
     channelLogger: ChannelLogger,
     samCliContext: SamCliContext = getSamCliContext(),
     activationLaunchPath: ActivationLaunchPath = new ActivationLaunchPath()
-): Promise<CreateNewSamApplicationResults> {
-    const results: CreateNewSamApplicationResults = {
-        reason: 'unknown',
-        result: 'fail',
-        runtime: 'unknown'
-    }
+): Promise<void> {
+    let createResult: result = 'Succeeded'
+    let reason: createReason = 'unknown'
+    let createRuntime: runtime = 'unknown'
```

Comment:
Maybe `n/a` instead of `unknown`

Reply:
It's not `n/a` because are are always working with some runtime, we just don't know at this point

Reply:
Correct me if I'm wrong, but the only time `unknown` shows up is if the workflow is exited or fails prior to selecting a runtime? In which case, `n/a` would be proper.